packages/launchpad-auth/src/index.ts

```diff
@@ -71,21 +84,24 @@ async function getValidLaunchpadTokenFromS3() {
  * @param {string} params.passphrase - the passphrase of the Launchpad PKI certificate
  * @param {string} params.certificate - the name of the Launchpad PKI pfx certificate
  *
- * @returns {Promise.<string>} - the Launchpad token
+ * @returns {Promise<string>} - the Launchpad token
  *
  * @async
  * @alias module:launchpad-auth
  */
-async function getLaunchpadToken(params) {
+async function getLaunchpadToken(params: LaunchpadTokenParams): Promise<string> {
   let token = await getValidLaunchpadTokenFromS3();
 
   if (!token) {
     log.debug('getLaunchpadToken requesting launchpad token');
     const launchpad = new LaunchpadToken(params);
-    const tokenObject = await launchpad.requestToken();
-
+    const tokenResponse = await launchpad.requestToken();
     // add session_starttime to token object, assume token is generated 60s ago
-    tokenObject.session_starttime = (Date.now() / 1000 - 60);
+    const tokenObject:TokenObject = {
+      ...tokenResponse,
+      session_starttime: (Date.now() / 1000 - 60)
```

Comment:
This isn't _wrong_, but it isn't obviously right either unless you're real sure you remember the order of operations.

```suggestion
      session_starttime: (Date.now() / 1000) - 60
```